index.js

```diff
@@ -64,6 +64,72 @@ let getRouteData = ({options: {action}, _sails: {_actions}}) => ({
   action: _.get(`${action}.name`, _actions) || _.last(action.split('/'))
 })
 
+let methodAliases = {
+  createRecord: 'create',
+  updateOneRecord: 'update',
+  destroyOneRecord: 'destroy',
+  findRecords: 'find',
+}
+let getAction = action => _.get(action, methodAliases) || action
+let routeInfo = req => {
+  let { action, type } = getRouteData(req)
+  return {
+    type,
+    action: getAction(action),
+    Model: req._sails.models[type],
+  }
+}
+
+// modelPolicy will trigger any of the policies in the model based on the CRUD actions that are called from the client.
+// modelPolicy will also filter properties by isDeleted by default, which provides an easy way to deal with soft-deleted records.
+let modelPolicy = policy(async (req, res, params = {}) => {
+  let { action, Model } = routeInfo(req)
+  if (/find/i.test(action)) {
+    if (!req.body) req.body = {}
+    params.isDeleted = req.body.isDeleted = { '!=': true }
+  }
+  return _.getOr(_.noop, `policies.${action}`, Model)(req, res, params)
+})
+
+// modelPostPolicy is the same as the modelPolicy, but looks for properties in postPolicy (which are treated as postPolicies)
+let modelPostPolicy = postPolicy(async (req, res, response) => {
+  let { action, Model } = routeInfo(req)
+  return _.getOr(_.noop, `postPolicies.${action}`, Model)(req, res, response)
+})
+
+// populatePolicy and populatePostPolicy go together.
+// populatePolicy rips out any `populate` property coming from the request params or body,
+// then sets this object in the request itself. This object Should contain as properties paths
+// indicating where the ID properties are, and as value the model that will be
+// used to fetch the right values. For example:
+//     { 'path/to/userID': 'user' }
```

Comment:
is there a reason why we are using / instead of a dot notation, to me a dot notation feels more natural for what we are trying to accomplish here, this is not a url or anything like that